### PR TITLE
Refactor types and make array of struct work with ts

### LINKED ISSUE #6827: Types: Array of structs not working
```
td struct bar { int a; int b; };
td struct foo { int x; struct bar moo[5]; }
```

Similar to https://github.com/radare/radare2/issues/6826

### PR DESCRIPTION
**Pull request by sivaramaaa**
Fixes #6827 

### Before fix

* Struct of Struct

```
[0x00000000]> "td struct foo { int a; int x ; };"
[0x00000000]> "td struct bar { int a; struct foo y ; };"
[0x00000000]> ts bar
pf d? a (foo)y
[0x00000000]> .ts bar
Cannot find format for struct `foo`
Cannot find format for struct `foo`
Undefned struct 'foo'.
a : 0x00000000  = 0
y :
```
* Array of struct 

```
[0x00000000]> "td struct foo { int a; int x ; };"
[0x00000000]> "td struct bar { int a; struct foo y[10] ; };"
[0x00000000]> ts bar
pf d? a (foo)y // was same as struct of struct
```

### After fix 

```
[0x00000000]> "td struct foo { int a; int b; };"
[0x00000000]> "td struct bar { int x; struct foo y[2]; };"
[0x00000000]> ts bar
pf d[2]? x (foo)y
[0x00000000]> .ts bar
 x : 0x00000000 = 0
 y : 
[
                struct<foo>
   a : 0x00000004 = 0
   b : 0x00000008 = 0

                struct<foo>
   a : 0x0000000c = 0
   b : 0x00000010 = 0
]
[0x00000000]>
```
Added tests in r2r [#1315](https://github.com/radare/radare2-regressions/pull/1315) too .

### CODE REVIEW

File: libr/core/cmd_type.c [hunk 1]
```diff
@@ -470,40 +471,41 @@ static int cmd_type(void *data, const char *input) {
...
 		switch (input[1]) {
 		case '?':
 			r_core_cmd_help (core, help_msg_ts);
 			break;
 		case '*':
-			showFormat (core, input + 2, 1);
+			showFormat (core, name + 1, 1);
 			break;
 		case ' ':
 			showFormat (core, input + 2, 0);
```
Comment (XVilka) [inline libr/core/cmd_type.c]: Maybe change here to `name` too?

**Comment by XVilka** [comment]
LGTM.

File: libr/core/cmd_type.c (hunk 1 above)
Comment (sivaramaaa) [reply libr/core/cmd_type.c]: No since that case is a space already , name would be null there !

Comment (sivaramaaa) [reply libr/core/cmd_type.c]: Yup done !

**Comment by radare** [comment]
awesome <3

**Pull request merged**